nil/principles.mdx

```diff
@@ -0,0 +1,64 @@
+import Flows from '@site/static/img/nil/flows.png'
+
+# Principles
+
+This page offers a high-level explanation of the architecture of =nil;.
+
+## Flows
+
+The following diagram shows how =nil; organizes execution shards, the main shard, the transfer of ZKPs, and communications with Ethereum.
+
+<img src={Flows}/ >
+
+In brief, each execution shard is responsible for generating a ZKP for their respective state change. Then the proofs are verified via a [**HotStuff2-based**](https://eprint.iacr.org/2023/397.pdf) global consensus protocol. 
+
+The main shard uses all received ZKPs to create a 'master' ZKP that is then transmitted to Ethereum. The global state of =nil; is verified and transactions reach finality.
+
+## Consensus shard
+
+The consensus shard performs two essential functions:
+
+* The shard stores and sets consensus rules and parameters
+* The shard stores hashes of the most recent blocks from execution shards and ensures synchronization between execution shards
+
+:::info
+
+There is only one consensus shard in the cluster.
+
+:::
+
+## Execution shards
+
+As implied by their name, execution shards process transactions. 
+
+Each shard is allocated with a subset of accounts from the global state. The shard is responsible for executing transactions made to and from these accounts. Consensus is handled by a committee made up of several validators.
+
+To learn more about execution shards, [**click here**](./core-concepts/shards-parallel-execution#execution-shards).
+
+:::info[Block capacity]
+
+Execution shards have a preset block capacity determined by the consensus shard. If a shard reaches capacity, it is split into two. If at least half of the available capacity is vacant, the shard is merged with another shard.
```

Comment:
Should be updated according to the last changes

Reply:
info about capacity/splits/merges is removed